public/app/percona/shared/core/reducers.ts

```diff
@@ -394,6 +395,56 @@ export const fetchServerSaasHostAction = createAsyncThunk(
     )
 );
 
+export interface UserDetails {
+  userId: number;
+  productTourCompleted: boolean;
+}
+
+const userDetailsInitialState: UserDetails = {
+  userId: 0,
+  productTourCompleted: false,
+};
+
+const toUserDetailsModel = (res: UserDetailsResponse): UserDetails => ({
+  userId: res.user_id,
+  productTourCompleted: !!res.product_tour_completed,
+});
+
+export const userDetailsSlice = createSlice({
```

Comment:
I understand this approach. But we already have a `perconaUserSlice`, we can probably extend that and `PerconaUserState` to allow these two properties (userId, productTourCompleted) and add a new reducer to the slice that sets these two (or two reducers that set them separately).

Reply:
Yup, makes more sense to extend it.